living_documentation_generator/generator.py

```diff
@@ -0,0 +1,316 @@
+        issue_md_page_content = issue_page_template.format(**replacements)
+
+        # Save the single issue Markdown page
+        page_filename = consolidated_issue.generate_page_filename()
+        with open(os.path.join(self.output_path, page_filename), 'w', encoding='utf-8') as issue_file:
+            issue_file.write(issue_md_page_content)
+
+        print(f"Generated {page_filename}.")
+
+    def _generate_index_page(self, issue_index_page_template: str,
+                             consolidated_issues: dict[str, ConsolidatedIssue]):
+        issue_table = """| Organization name     | Repository name | Issue 'Number - Title'  | Linked to project | Project status  |Issue URL   |
+               |-----------------------|-----------------|---------------------------|---------|------|-----|
+               """
+
+        # Create an issue summary table for every issue
+        for key, consolidated_issue in consolidated_issues.items():
+            issue_table += self._generate_markdown_line(consolidated_issue)
+
+        # Prepare issues replacement for the index page
+        replacement = {
+            "date": datetime.now().strftime("%Y-%m-%d"),
+            "issues": issue_table
+        }
+
+        # Replace the issue placeholders in the index template
+        index_page = issue_index_page_template.format(**replacement)
+
+        # Create an index page file
+        with open(os.path.join(self.output_path, "_index.md"), 'w', encoding='utf-8') as index_file:
+            index_file.write(index_page)
+
+        logging.info("Generated _index.md.")
+
+    def _generate_markdown_line(self, consolidated_issue) -> str:
+        """
+            Generates a markdown summary line for a given feature.
+
+            @param consolidated_issue: The dictionary containing feature data.
+
+            @return: A string representing the markdown line for the feature.
+        """
+
+        # Extract issue details from the consolidated issue object
+        organization_name = consolidated_issue.organization_name
+        repository_name = consolidated_issue.repository_name
+        number = consolidated_issue.number
+        title = consolidated_issue.title
+        title = title.replace("|", " _ ")
+        page_filename = consolidated_issue.generate_page_filename()
+        status = consolidated_issue.status
+        url = consolidated_issue.url
+
+        # Change the bool values to more user-friendly characters
+        if consolidated_issue.linked_to_project:
+            linked_to_project = "🟢"
+        else:
+            linked_to_project = "🔴"
+
+        # Generate the markdown line for the issue
+        md_issue_line = (f"|{organization_name} | {repository_name} | [#{number} - {title}]({page_filename}) |"
+                         f" {linked_to_project} | {status} |[GitHub link]({url}) |\n")
+
+        return md_issue_line
+
+    def generate_issue_summary_table(self, consolidated_issue: ConsolidatedIssue) -> str:
```

Comment:
Should be private / protected method as other methods.

Reply:
Addressed in commit = TBD.